xla/stream_executor/device_memory.h

```diff
@@ -96,7 +96,11 @@ class DeviceMemoryBase {
   }
 
  private:
-  void *opaque_;  // Platform-dependent value representing allocated memory.
+  // Platform-dependent value representing allocated memory.
+  // User may also constructs the boject with null opaque_pointer and non-zero
+  // size, which indicates the case that buffer is allocated lazily through
+  // command buffer;
+  void *opaque_;  
```

Comment:
can we use a special address instead of nullptr? Something that NVIDIA driver will never allocate itself? `0xFFFFFFFF`? `0xDEADBEEF`? 

All addresses are 32 bytes aligned? Can `0x0000000F` all lower bits set to `1` be used as a lazy marker?

Reply:
And use fall back on se::CommandBuffer in GetDeviceMemory only if address is indeed a lazy marker